src/server/handlers.rs

```diff
@@ -521,6 +521,102 @@ impl<P: Publisher> Handler<P> {
                 }
             };
 
+        // Retrieve all stored models in the lattice
+        let stored_models = match self.store.list(account_id, lattice_id).await {
+            Ok(d) => d,
+            Err(e) => {
+                error!(error = %e, "Unable to fetch data");
+                self.send_error(msg.reply, "Internal storage error".to_string())
+                    .await;
+                return;
+            }
+        };
+
+        // Retrieve all the existing provider refs in store that are currently deployed
+        let mut existing_provider_refs: HashMap<String, String> = HashMap::new();
+        for model_summary in stored_models.iter() {
+            if let Some(deployed_version) = &model_summary.deployed_version {
+                let (stored_manifest, _) = match self
+                    .store
+                    .get(account_id, lattice_id, &model_summary.name)
+                    .await
+                {
+                    Ok(Some(m)) => m,
+                    Ok(None) => (StoredManifest::default(), 0),
+                    Err(e) => {
+                        error!(error = %e, "Unable to fetch data");
+                        self.send_error(msg.reply, "Internal storage error".to_string())
+                            .await;
+                        return;
+                    }
+                };
```

Comment:
I think if you use the suggestion in the above comment with `get_deployed` you can get rid of this check altogether 😄

Reply:
Modifying the check instead of removing since we don't want to perform a fetch operation if there is no currently deployed version.